### DISCUSSION TITLE
Can I run a script on entering and leaving a particular workspace?

### DISCUSSION THREAD

**Discussion by emmalexandria** [i3/i3#4733] [Q&A]
I have a laptop with a very puny battery, and Firefox draws about as much as all of Xorg when running. I want to write a shellscript to send SIGSTOP to Firefox when I leave my browser workspace, and then send a SIGCONT when I enter it again to conserve battery life. I could probably keybind this behaviour and do it manually, but I'd prefer to automate it.

**Comment by Airblader** [comment]
You can, but not through i3's configuration surface. You'll need to write a script, e.g. using the IPC we expose to listen for workspace focus events and checking the tree where Firefox is situated. You can also do this without the IPC using EWMH atoms directly for a more window manager agnostic approach, but using the IPC is probably easier as it will not require understanding of the protocols.
